src/analysis/special_functions/gamma.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import measure_theory.integral.exp_decay
+
+/-!
+# The Gamma function
+
+This file treats Euler's integral for the `Γ` function, `∫ x in Ioi 0, exp (-x) * x ^ (s - 1)`, for
+`s` a real or complex variable.
+
+We prove convergence of the integral for `1 ≤ s` in the real case, and `1 ≤ re s` in the complex
+case (which is non-optimal, but the optimal bound of `0 < s`, resp `0 < re s`, is harder to prove
+using the methods in the library). We also show `Γ(1) = 1`.
+
+The recurrence `Γ(s + 1) = s * Γ(s)`, holomorphy in `s`, and extension to the whole complex plane
+will be added in future pull requests.
+
+## Tags
+
+gamma
+-/
+
+noncomputable theory
+open filter interval_integral set real measure_theory
+open_locale topological_space
+
+section gamma_real
+
+/-- Asymptotic bound for the Γ function integrand. -/
+lemma gamma_integrand_is_O (s : ℝ) : asymptotics.is_O (λ x:ℝ, exp (-x) * x ^ s)
+  (λ x:ℝ, exp (-(1/2) * x)) at_top :=
+begin
+  refine asymptotics.is_o.is_O (asymptotics.is_o_of_tendsto _ _),
+  { intros x hx, exfalso, exact (exp_pos (-(1 / 2) * x)).ne' hx },
+  have : (λ (x:ℝ), exp (-x) * x ^ s / exp (-(1 / 2) * x)) = (λ (x:ℝ), exp ((1 / 2) * x) / x ^ s )⁻¹,
+  { ext1 x, dsimp,
+    have : exp (-x) = exp (-(1 / 2) * x) * exp (-(1 / 2) * x),
+    { rw ←real.exp_add, field_simp },
+    rw this,
+    have : exp (1 / 2 * x) = (exp (-(1 / 2) * x))⁻¹ := by { rw ←exp_neg, field_simp, },
+    rw this,
+    field_simp [(exp_pos (-x/2)).ne'], ring },
```

Comment:
```suggestion
  { ext1 x, 
    field_simp [exp_ne_zero, exp_neg, ← real.exp_add],
    left,
    ring },
```
Even better, one should probably add the attribute `@[field_simps]` to `real.exp_ne_zero` (and also to the complex version), and then you don't even need to mention it on the `field_simp` line.

Reply:
@sgouezel , your code adds a non-final `simp` (more precisely, `field_simp`).

Reply:
`field_simp` followed by `ring` is supposed to be ok, right? Granted, there is a `left` in the middle, but there is nothing nontrivial like an `apply` or an `rw`, so it looks robust enough to me.